### PR TITLE
Add check if user has permission to add a project to a profile.

### PR DESCRIPTION
**Pull request by ukanga**

### CODE REVIEW

**Comment by pld** [comment]
is there an issue this is tied to?

**Comment by moshthepitt** [comment]
@pld I believe this is tied to https://github.com/onaio/core/issues/1357

File: onadata/apps/api/management/commands/apply_can_add_project_perms.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+"""
+Command apply_can_add_preject_perms - applys can_add_project permission to all
```
Comment (pld) [inline onadata/apps/api/management/commands/apply_can_add_project_perms.py]: typo, `preject` -> `project`

File: onadata/apps/api/management/commands/apply_can_add_project_perms.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    for user in users.iterator():
+        permissions = user.userprofileuserobjectpermission_set.filter(
+            permission__codename='can_add_xform')
+        for permission in permissions:
+            assign_perm('can_add_project', permission.user, user)
+
+
+class Command(BaseCommand):
+    """
+    Command apply_can_add_preject_perms - applys can_add_project permission to
```
Comment (pld) [inline onadata/apps/api/management/commands/apply_can_add_project_perms.py]: typo

File: onadata/libs/serializers/project_serializer.py [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
-    PROJECT_LINKED_DATAVIEWS, PROJ_BASE_FORMS_CACHE)
-from onadata.apps.api.tools import (
-    get_organization_members_team, get_organization_owners_team)
+    PROJ_BASE_FORMS_CACHE, PROJ_FORMS_CACHE, PROJ_NUM_DATASET_CACHE,
+    PROJ_PERM_CACHE, PROJ_SUB_DATE_CACHE, PROJ_TEAM_USERS_CACHE,
+    PROJECT_LINKED_DATAVIEWS, safe_delete)
+from onadata.libs.utils.decorators import check_obj
 
 
 def get_obj_xforms(obj):
```
Comment (pld) [inline onadata/libs/serializers/project_serializer.py]: I don't remember if there's a historical reason why we have `obj` throughout this namespace, but it looks like a future refactor would be to change that to `project`

File: onadata/libs/serializers/project_serializer.py [hunk 4]
```diff
@@ -158,55 +181,47 @@ class Meta:
...
-            'is_merged_dataset',
-        )
-
-    def get_published_by_formbuilder(self, obj):
-        md = obj.metadata_set.filter(
-            data_type='published_by_formbuilder'
-        ).first()
-        if md and hasattr(md, 'data_value') and md.data_value:
-            return True
+        fields = ('name', 'formid', 'id_string', 'num_of_submissions',
```
Comment (pld) [inline onadata/libs/serializers/project_serializer.py]: 👎

**Comment by ukanga** [comment]
> But also, I don't see any explicit tested for project creation in another users account that replicates the production error.

@pld are you talking about this test? https://github.com/onaio/onadata/blob/387212f5996b9497dbd5ac7523a3a469d0009805/onadata/apps/api/tests/viewsets/test_project_viewset.py#L1500

**Comment by pld** [comment]
No, since that's tests for moving a project. Replicating the error would be on creating a project in another users account.

**Comment by pld** [comment]
nearly green, let's merge, get it in next release

**Pull request merged**